Assets/Plugins/Clean/Tests/ComponentUpgradeTests.cs

```diff
@@ -0,0 +1,142 @@
+
+            return prefab;
+        }
+
+        public string Setup()
+        {
+            var guid = Guid.NewGuid().ToString();
+            if (!Directory.Exists($"Assets/{guid}")) ;
+            {
+                AssetDatabase.CreateFolder("Assets", guid);
+            }
+            return guid;
+        }
+
+        public void Cleanup(string guid)
+        {
+            if (Directory.Exists($"Assets/{guid}")) ;
+            FileUtil.DeleteFileOrDirectory($"Assets/{guid}");
+            if (File.Exists($"Assets/{guid}.meta"))
+                FileUtil.DeleteFileOrDirectory($"Assets/{guid}.meta");
+        }
+
+        [Test]
+        public void CanUpgradeTextComponent()
+        {
+            var testText = "some text";
+            var textComponent = CreateText(testText);
+            var go = textComponent.gameObject;
+
+            ComponentUpgrade.UpgradeText(textComponent);
+
+            var tmp = go.GetComponent<TextMeshProUGUI>();
+            Assert.IsNotNull(tmp, "Failed to add TextMeshProUGUI component");
+            Assert.IsNull(tmp.GetComponent<Text>(), "Failed to remove Text component");
+            Assert.AreEqual(tmp.text, testText, "Failed to update text field");
+        }
+
+        [Test]
+        public void CanUpgradeTextPrefab()
+        {
+            var guid = Setup();
+            var prefabPath = $"Assets/{guid}/text.prefab";
+            GameObject tmpInstance = null;
+
+            try
+            {
+                // Create prefab
+                CreateTextPrefab("original text", prefabPath);
+
+                // ComponentUpgrade prefab
+                ComponentUpgrade.UpgradePrefabRoot(prefabPath);
+
+                // Instantiate clone
+                tmpInstance = PrefabUtility.LoadPrefabContents(prefabPath);
+                var tmp = tmpInstance.GetComponent<TextMeshProUGUI>();
+
+                // Check clone
+                Assert.IsNull(tmpInstance.GetComponent<Text>(),
+                    "Failed to remove Text component from prefab instance");
+                Assert.IsNotNull(tmpInstance.GetComponent<TextMeshProUGUI>(),
+                    "Failed to add TextMeshProUGUI component to existing prefab instance");
+                Assert.IsNotNull(tmp, "Failed to add TextMeshProUGUI component to new prefab instance");
+                Assert.AreEqual(tmp.text, "original text");
+            }
+            finally
+            {
+                if (tmpInstance != null) PrefabUtility.UnloadPrefabContents(tmpInstance);
+                Cleanup(guid);
+            }
+        }
+
+        [Test]
+        public void CanUpgradePrefabWithSceneOverrides()
+        {
+            var guid = Setup();
+            try
+            {
+                // Setup scene override
+                var prefabPath = $"Assets/{guid}/text.prefab";
+                var prefab = CreateTextPrefab("original text", prefabPath);
```

Comment:
Can't we actually do this with a theory where the text string is generated at random?